src/peft/utils/other.py

```diff
@@ -498,3 +498,23 @@ def id_tensor_storage(tensor: torch.Tensor) -> Tuple[torch.device, int, int]:
         unique_id = storage_ptr(tensor)
 
     return tensor.device, unique_id, storage_size(tensor)
+
+
+def cast_non_trainable_to_dtype(model, dtype):
```

Comment:
Just noticing that we may want to use a different name, since the trainable parameters will also be cast. The name could give the impression that trainable parameters are left untouched.

Reply:
Updated the PR to use `cast_mixed_precision_params`, let me know if you have better naming suggestions.